packages/git/src/browser/dirty-diff/dirty-diff-manager.ts

```diff
@@ -239,7 +239,7 @@ export class DirtyDiffModel implements Disposable {
     protected async isInGitRepository(repository: Repository): Promise<boolean> {
         const modelUri = this.editor.uri.withoutScheme().toString();
         const repoUri = new URI(repository.localUri).withoutScheme().toString();
-        return modelUri.startsWith(repoUri) && await this.previousRevision.isVersionControlled();
+        return modelUri.startsWith(repoUri) && this.previousRevision.isVersionControlled();
```

Comment:
Await should stay in this case
```ts
    export interface PreviousFileRevision {
        readonly fileUri: URI;
        getContents(staged: boolean): Promise<string>;
        isVersionControlled(): Promise<boolean>;
    }
```

Reply:
I wasn't sure, but after trying it, it works as intended.  An expression "a_boolean && a_boolean_promise" seems to actually wait for the promise result.

Reply:
Ah I see, the `&&` operator will return the left part if evaluated as `false`, otherwise the right part, which then gets wrapped by the outer async function. Phew.

Reply:
For the record (I talked with @marechal-p offline), the presence or not of the await keyword can change the result, depending on the order of the condition:

```
async function paul1() {
    const p1 = new Promise((resolve) => setTimeout(() => resolve(false), 1));
    return p1 && true;
}
async function paul2() {
    const p1 = new Promise((resolve) => setTimeout(() => resolve(false), 1));
    return await p1 && true;
}
async function paul3() {
    const p1 = new Promise((resolve) => setTimeout(() => resolve(false), 1));
    return true && p1;
}
async function paul4() {
    const p1 = new Promise((resolve) => setTimeout(() => resolve(false), 1));
    return true && await p1;
}

paul1().then((value) => console.log("Result 1 is", value));  // true
paul2().then((value) => console.log("Result 2 is", value));  // false
paul3().then((value) => console.log("Result 3 is", value));  // false
paul4().then((value) => console.log("Result 4 is", value));  // false
```

However, tslint is "smart" enough to only suggest the removal of `await` in the latest case (`paul4`).  It doesn't suggest it for `paul2`.  This means that as far as I know, we can trust tslint for this :).